doc/guides/build_tools.rst

```diff
@@ -0,0 +1,107 @@
+.. _build_tools:
+
+Build tools
```

Comment:
Those are not build tools and having a guide called build tools is misleading. Maybe it should be called the "Zephyr Build Process" and provide more details about how those are invoked and when. We should not give the impression that a developer needs to invoke them.

Reply:
How about "Internal build tools"?  @andrewboie can you give me a better intro paragraph?

Reply:
> Considering they're "internal" and never invoked directly, do their pydocs even need to be published on the web site?
> 
> By the way one section of the documentation is currently called "User and Developer guides" but it doesn't seem to draw a line between a "User" and a "Developer". I haven't seen any line either between 1. an app developer (who shouldn't care about these scripts, black box) versus 2. a core Zephyr developer who may find useful gory implementation details about these scripts or about cmake (PR #13805).

let's not digress here. The documentation is needed, those however are neither "external" nor "internal" build tools. We need the documentation to be about the build process and what steps we take to generate the final binary, so there is a leading section that is missing here and probably some workflow diagram describing the build process. 
AFAIK we have this information and documentation, we just need to create some content around the text coming from the scripts.